lib/rubocop/path_util.rb

```diff
@@ -28,14 +28,19 @@ def relative_path(path, base_dir = Dir.pwd)
         end
     end
 
+    SMART_PATH_CACHE = {} # rubocop:disable Style/MutableConstant
```

Comment:
Since this is not the first such disabling in the codebase, wdyt on supporting `AllowedPatterns` for `Style/MutableConstant` cop?

Reply:
@bbatsov Updated this PR. In this case we cannot use a module instance variable (`@smart_path_cache`), because this module additionally to just being used directly, is also included into other classes. So we will get an undefined instance variable in this case.
Unless I am missing something.